### DISCUSSION TITLE
Support Multiple MySQL Schemas

### DISCUSSION THREAD

**Discussion by totalhack** [directus/directus#16786] [Feature Requests]
I am new to Directus and was *very* surprised to see that it can not handle multiple schemas in MySQL, which seems like a pretty reasonable base-level requirement. I've seen all of the tickets and discussion around PostgreSQL and the use of DB_SEARCH_PATH, and also [comments](https://github.com/directus/directus/issues/4411#issuecomment-846329731) that suggest it might be a feature coming down the road, but that was some time ago and seems to focus on PostgreSQL.

This more or less makes Directus unusable for all but the simplest of existing MySQL databases unless I want to add/manage logic to propagate/duplicate data between Directus collections and the tables they are meant to target in other schemas. 

Is there an updated ETA on support for multiple schemas? 

Are there workarounds I may have missed?

**Comment by rijkvanzanten** [comment]
Am I mistaken that in MySQL "schema" is synonymous with "database"? Are you asking to support multiple MySQL databases at a time?

> I am new to Directus and was very surprised to see that it can not handle multiple schemas in MySQL, which seems like a pretty reasonable base-level requirement

You'd be surprised. In the years this project has existed, you're the first to mention it!

> Is there an updated ETA on support for multiple schemas?

We prioritize / implement feature requests based on community engagement and popularity, and as mentioned above, this hasn't been brought up at all before! Choosing a schema to use on postgres/the other DB vendors that have native schema support is a feature that has been requested and something we're looking to support in the future. Supporting multiple databases simultaneously is similarly something we're looking into, but I'm not sure that solves for the same thing you're asking for.

**Comment by totalhack** [comment]
Yes, apologies, database and schema can be unclear between different contexts and DB technologies, but it's technically creating a "database" in MySQL, though I've often seen these referred to as separate schema since database can mean "the whole MySQL database."  

Not sure how common MySQL usage is for Directus into an existing DB (my use case was to manage existing tables, hence the issue), but I know PostgreSQL is a much more popular target with open source projects in recent years so I'm not completely surprised. 

Anyway, part of my surprise is that this should be a really easy thing to support logically speaking, unless there were some design decisions made in the early days that make that a pain to change now. How is one supposed to take advantage of Directus capabilities to manage existing tables if everything needs to be in the Directus schema? Maybe that's just not a use case you are trying to support?

**Reply by rijkvanzanten** [reply]
> Anyway, part of my surprise is that this should be a really easy thing to support logically speaking

Bold statement! Directus supports most flavors of SQL database, so it's not just a trivial little snippet to add to magically enable multiple databases in mysql (we would've done it already otherwise!). Feel free to poke around the codebase to get a feel for the level of complexity this really entails in a project like this 🙂

**Reply by totalhack** [reply]
I appreciate the offer, but will have to pass. After all that's why I said the part about early design decisions, I'm sure if it were easy now you would do it. All I know from the outside looking in is it's something many other libraries and tools handle without issue. 

This is going nowhere so time to call it. Good luck out there!

**Reply by rijkvanzanten** [reply]
> All I know from the outside looking in is it's something many other libraries and tools handle without issue.

Could you share some that came to mind? I'm curious what we can learn from their implementations. I haven't seen this myself, but that's very much coming from a front-end dev / cms world 🙂

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
